File: DSA/SearchingAndSorting/quick-sort.py
def quicksort(array):
    if len(array) <= 1:
        return array

    pivot = array[0]
    left = []
    right = []

    for i in range(1, len(array)):
        if array[i] < pivot:
            left.append(array[i])
        else:
            right.append(array[i])

    sortedLeft = quicksort(left)
    sortedRight = quicksort(right)

    return sortedLeft + [pivot] + sortedRight


test = [21, 4, 1, 3, 9, 20, 25, 6, 21, 14]
print(quicksort(test))

File: DSA/SearchingAndSorting/quickSort.js
const quickSort = (array) => {
	if (array.length <= 1) {
		return array
	}

	const pv = array[0]
	const left = []
	const right = []

	for (let i = 1; i < array.length; i++) {
		if (array[i] < pv) {
			left.push(array[i])
		} else {
			right.push(array[i])
		}
	}

	const sortedLeft = quickSort(left)
	const sortedRight = quickSort(right)

	return sortedLeft.concat(pv, sortedRight)
}

test = [21, 4, 1, 3, 9, 20, 25, 6, 21, 14]
console.log(quickSort(test))
